src/main/java/seedu/address/storage/JsonSerializableAddressBook.java

```diff
@@ -81,8 +83,10 @@ public AddressBook toModelType() throws IllegalValueException {
 
         for (JsonAdaptedJob jsonAdaptedJob : jobs) {
             Job job = jsonAdaptedJob.toModelType();
-            // TODO: Duplicate job invalidation
-            addressBook.addJob(job);
+            if (addressBook.hasJob(job)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_JOB);
+            }
+            addressBook.addJob(job); // this can throw CompanyNotFoundException
```

Comment:
I touched these exact lines in my PR XD

Reply:
I just saw it was TODO and it was quite simple so I implemented it lol